ls1intum/Artemis
# `Development`: Improve file input validation

### Checklist
#### General


#### Server

### Motivation and Context


Improve the methods surrounding input validation and file saves.

### Description
We improved the name of several methods to reflect the functionality better, extracted the functionality of the file save method to improve readability and added appropriate tests for extracted methods.

### Steps for Testing
We just want to make sure that I didn't accidentally break the implementation.

Prerequisites:
- 1 Instructor, 1 Student

Split Lecture PDF (Instructor):
1. Open the lecture tab and click on "Create Lecture"
4. Fill in the title and channel name
5. Check "Automatic unit processing" at the bottom
7. Select a multipage PDF to process and hit "Process units". If you use a PDF that fits the requirements, skip point 5. Otherwise, continue.
8. Add units manually, at least two and set the pages accordingly. Then hit "submit.
9. Open the lecture and download a lecture unit attachment.
11. No issues should occur

Fileupload Exercise (Instructor, Student):
1. Create a file upload exercise with a description. Upload an image to the description
2. Start the exercise, the image should show up properly as well as the rest of the description
3. Participate in the exercise
4. Upload another file
5. View/Download your submission
6. No issues should occur, the correct file should download.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| FileService.java | 85.4% | ✅              |
